SpringProject/src/main/java/com/gradproject2019/users/service/UserServiceImpl.java

```diff
@@ -122,6 +123,36 @@ private void checkTokenMatchesUser(UUID token, Long userId) {
         }
     }
 
+    private void checkIdMatchesManagerId(User requestingUser, User requestedUser) {
+        if(!requestingUser.getId().equals(requestedUser.getManagerId())) {
+            throw new UserUnauthorisedException();
+        }
+    }
+
+    private void checkManagerIdMatchesId(User requestingUser, User requestedUser) {
+        if(!requestingUser.getManagerId().equals(requestedUser.getId())) {
+            throw new UserUnauthorisedException();
+        }
+    }
+
+    private void checkUserRequestingIsAuthorized(Long requestedUserId, UUID token) {
+        User requestingUser = getUserById(authServiceImpl.getTokenById(token).getUserId());
+        User requestedUser =getUserById(requestedUserId);
+        try {
+            checkTokenMatchesUser(token, requestedUserId);
+        } catch (UserUnauthorisedException a) {
+            try {
+                checkIdMatchesManagerId(requestingUser, requestedUser);
+            } catch (UserUnauthorisedException | NullPointerException b) {
+                try {
+                checkManagerIdMatchesId(requestingUser, requestedUser);
+                } catch (NullPointerException c) {
+                    throw new UserUnauthorisedException();
+                }
+            }
+        }
+    }
+
```

Comment:
I realise this is disgusting, I have a solution involving if statements and negating the need to catch null pointers which I thought of as I was leaving yesterday. This will be implemented on Monday next week.

Reply:
This has been fixed in the latest commit.